tests/test_depsolver.py

```diff
@@ -361,15 +363,14 @@ def _prepare_test_data(pulp):
             RpmDependency(name="/some/script"),
         ],
     )
-    # note: the dependency "/some/script" will be skipped from processing
 
     unit_3 = RpmUnit(
         name="apr",
         version="100",
         release="200",
         epoch="1",
         arch="x86_64",
-        provides=[RpmDependency(name="apr")],
+        provides=[RpmDependency(name="apr"), RpmDependency(name="/some/script")],
```

Comment:
@rbikar, it can't be this easy, right? The docstring for RpmDependency name attr says it can be for a scriptlet but do we really expect RPM units to list requires/provides as seen here, as filepaths?

I suspect we'll actually need to query RpmUnits with files.file matching the filepath, which likely means changes to RpmDependency and/or RpmUnit, but I can't confirm that field's existence. I've spent quite a while digging in rcm-pulp ("rhsm-pulp" doesn't seem to exist in Github, Gitlab or Gerrit) and I can't find models or schema for units.

Reply:
No, it's not that easy.

A) Related JIRA already has the detail how to query. You may want to give it a try on real data.
B) Why do you think you need changes in RpmDependency and/or RpmUnit? It shouldn't be needed for this purpose. It should suffice to query `files.file` and work with returned units. If you think it'll be too cumbersome  (I guess it may be) that you can add another field to `RpmUnit` is quite straightforward. I admit that the solution with new field on `RpmUnit` holding the file list would probably be more elegant. 

So typically the scriptlet are present in requires section (as `RpmDependency.name`) but they are not stored in provides but in field `files.file` which holds list of files provides by rpm.

Reply:
Thanks. Submitted https://github.com/release-engineering/pubtools-pulplib/pull/220